content/articles/getting-started-with-hookstate/index.md

```diff
@@ -0,0 +1,186 @@
+
+    // we can filter task lists to remove a list item
+    removeTask(id) {
+      return state.set((tasks) => tasks.filter((task) => task.id !== id));
+    },
+
+    // state.get() retieves the Todo list state 
+    get getTasks() {
+      return state.get();
+    }
+  };
+}
+```
+
+To manage a global state, Hookstate provides the `get` method and `set` methods on the newly created state to access and manipulate it. The `uuid` module generates a random ID for each item on the to-do list.
+
+### Create `AddTodo` component
+
+Define the `AddTodo.jsx` file inside the `components` folder. Our component will reuse the previous custom `useTaskState` to access global state:
+
+```js
+import { useTaskState } from "../states/TaskState";
+```
+
+The `AddTodo` component returns JSX with a form to add a new to-do item. Create the `AddTodo` function:
+
+```js
+const AddTodo = () => {
+  const taskState = useTaskState();
+  return (
+    <div>
+      <form
+        onSubmit={(e) => {
+          e.preventDefault();
+          if (!e.target["toDo"].value.trim()) {
+            return;
+          }
+          taskState.addTask(e.target["toDo"].value);
+          e.target["toDo"].value = "";
+        }}
+      >
+	{/*An input element and a button to add a to-do item.*/}
+        <input name="toDo" />
+        <button type="submit">Add Todo</button>
+      </form>
+    </div>
+  );
+};
+```
+
+Finally, and export it on the same line:
+```js
+// Export component: AddTodo
+export default AddTodo;
+```
+
+### The `TodoList` component
+
+In our `TodoList` component, we will loop over our array of to-do lists and display them. Check the entire code below.
+
+```js
+import React from "react";
+import { useTaskState } from "../states/TaskState";
+
+const ToDoList = () => {
+  const taskState = useTaskState();
+  const state = taskState.getTasks;
+  return (
+      <ul>
+        {state.length > 0 &&
+          state.map((todo) => (
+            <li key={todo.id}>
+              <span>{todo.text}</span>
+              {/*Add a delete button*/}
+              <button onClick={() => taskState.removeTask(todo.id)}>
+                Delete
+              </button>
+            </li>
+          ))}
+      </ul>
+  );
+};
+
+export default ToDoList;
+```
+
+### The `App.js` Component
+Our main `App.js` file is quite minimal. All we need to add is the `ToDoList` component, `AddTodo` component, and `useTaskState`.
+
+```js
+import { useTaskState } from "./states/TaskState";
+import ToDoList from "./components/ToDoList";
+import AddTodo from "./components/AddTodo";
+```
+Finally, export the `App` component with the components mapped to the JSX.
+```js
+export default function App() {
+  const taskState = useTaskState();
+  // console.log(taskState.getTasks);
+  return (
+    <div className="App">
+      <h1>TODO APP</h1>
+
+      <AddTodo />
+      <ToDoList />
+    </div>
+  );
+}
+```
+
+### Demo
+![todo demo](/engineering-education/getting-started-with-hookstate/demo1.png)
```

Comment:
This looks like an abrupt end.

Make sure to end it neatly with proper conclusion.

Reply:
Explain how the user needs to run the above code and where to see the output.